src/assets/all_assets_request.rs

```diff
@@ -140,9 +124,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_limit(&mut self, limit: u32) -> &mut Self {
-        self.limit = Some(limit);
-        self
+    pub fn set_limit(self, limit: u32) -> Result<AllAssetsRequest, String> {
+        if limit < 1 || limit > 200 {
+            return Err("limit must be between 1 and 200".to_string());
+        }
```

Comment:
These are still runtime errors. We could introduce an 'InvalidAssetRequest' state and return this if the limit < 1 or > 200.

Reply:
Moved to https://github.com/Baseflow/stellar-rust-sdk/issues/8